docs/concepts/scheduler.md

```diff
@@ -37,8 +37,9 @@ The following **sequence** describes the steps involved to determine a seed cand
    * whose capacity for shoots would not be exceeded if the shoot is scheduled onto the seed, see [Ensuring seeds capacity for shoots is not exceeded](#ensuring-seeds-capacity-for-shoots-is-not-exceeded)
    * which are labelled with `seed.gardener.cloud/multi-zonal` if feature gate `HAControlPlanes` is turned on and shoot requests a high available control plane.
 1. Apply active [strategy](#strategies) e.g., _Minimal Distance strategy_
-1. Choose least utilized seed, i.e., the one with the least number of shoot control planes, will be the winner and written to the `.spec.seedName` field of the `Shoot`.
-
+1. Determine scores for seeds. The one with the least score will be the winner:
```

Comment:
> The one with the least score will be the winner:

The least score? 👀 This seems unintuitive. Shouldn't it be the other way round (highest score wins)?

Reply:
I think the issue is that `getSeedWithLeastShootsDeployed` is used as a scoring function. This function just returns the number of shoots. The bin packing algorithm used by default is to ensure equal distribution. So the one with the least number of shoots should be picked up first. That is why `least score` is used as a strategy

But i agree with you that it should be highest score if you are calling it `score` functions as that would be more intuitive. What can be done is change the implementation of `getSeedWithLeastShootsDeployed` to return `(1/<numberOfShoots>) * 1000` as the score value. The lower the usage, higher will be the score.

Reply:
Hm, I kind of thought the same but didn't want to bother to change the existing behavior which means that before also the one with a lower score (= shoot count) won the game.
I can think of several options:
- Don't call it score but rather penalty point (or maybe you have a better name), i.e. only change in terminology.
- Multiply by -1 so that the higher score wins.
- Design some scoring scheme so that less shoots will lead to a higher score.
Any other ideas or preferences? 😄

Reply:
I'm not sure whether this PR is the "right place" to introduce such scoring concepts. It seems this needs more thought and should probably discussed a bit upfront.
Maybe keep it simple for now and just don't consider zonal seeds for non-zonal shoots?

Reply:
>Maybe keep it simple for now and just don't consider zonal seeds for non-zonal shoots?

The only issue is that if all seeds in a region are `multi-zonal` then a `single-zonal` shoot control plane cannot be hosted on any multi-zonal seeds. If this can happen then one would need to choose a seed from the `multi-zonal` seed list as there are no `single-zone` seeds.

Reply:
Yes, I'm not proposing to keep it like this. However, given that this new feature gate was just introduced and will only be released for the first time with v1.49, the first implementation could be quite simple (as proposed above). I don't think that users will run into this situation right away, so we don't necessarily need a solution for v1.49. The scheduling can also be improved later on with more sophisticated solutions like scoring or whatever else. WDYT?

Reply:
Sorry @unmarshall I missed your comment. I think returning `1/<numberOfShoots>` can work perfectly, thanks for the suggestion.

@rfranzke considering that we'll have the release cut soon, does it make sense to postpone this PR in the meantime and file a "simple" one now? After the release I'd reopen this PR with the requested changes from you and @unmarshall. Others will have enough time to give their feedback as well.

Reply:
fine for me.

Reply:
Yes, sure, that's what I was suggesting in https://github.com/gardener/gardener/pull/6102#discussion_r896432168 😃

Reply:
Hm, I think it's not a big issue to have the scoring function as part

> Yes, sure, that's what I was suggesting in [#6102 (comment)](https://github.com/gardener/gardener/pull/6102#discussion_r896432168) 😃

Sorry, refreshing issue 😄